xma_data_processor.py

```diff
@@ -0,0 +1,374 @@
+            (frame_width, frame_height),
+        )
+        i = 1
+        while cam1_video.isOpened():
+            print(f"Current Frame: {i}")
+            ret_cam1, frame_cam1 = cam1_video.read()
+            _, frame_cam2 = cam2_video.read()
+            if ret_cam1:
+                frame_cam1 = cv2.cvtColor(frame_cam1, cv2.COLOR_BGR2BGRA, 4).astype(
+                    np.float32
+                )
+                frame_cam2 = cv2.cvtColor(frame_cam2, cv2.COLOR_BGR2BGRA, 4).astype(
+                    np.float32
+                )
+                frame_cam1 = cv2.normalize(
+                    frame_cam1, None, 0, 255, norm_type=cv2.NORM_MINMAX
+                )
+                frame_cam2 = cv2.normalize(
+                    frame_cam2, None, 0, 255, norm_type=cv2.NORM_MINMAX
+                )
+                if mode == "difference":
+                    extra_channel = blend_modes.difference(frame_cam1, frame_cam2, 1)
+                elif mode == "multiply":
+                    extra_channel = blend_modes.multiply(frame_cam1, frame_cam2, 1)
+                else:
+                    extra_channel = np.zeros((frame_width, frame_height, 3), np.uint8)
+                    extra_channel = cv2.cvtColor(
+                        extra_channel, cv2.COLOR_BGR2BGRA, 4
+                    ).astype(np.float32)
+                frame_cam1 = cv2.cvtColor(frame_cam1, cv2.COLOR_BGRA2BGR).astype(
+                    np.uint8
+                )
+                frame_cam2 = cv2.cvtColor(frame_cam2, cv2.COLOR_BGRA2BGR).astype(
+                    np.uint8
+                )
+                extra_channel = cv2.cvtColor(extra_channel, cv2.COLOR_BGRA2BGR).astype(
+                    np.uint8
+                )
+                frame_cam1 = cv2.cvtColor(frame_cam1, cv2.COLOR_BGR2GRAY)
+                frame_cam2 = cv2.cvtColor(frame_cam2, cv2.COLOR_BGR2GRAY)
+                extra_channel = cv2.cvtColor(extra_channel, cv2.COLOR_BGR2GRAY)
+                merged = cv2.merge((extra_channel, frame_cam2, frame_cam1))
+                out.write(merged)
+                if cv2.waitKey(1) & 0xFF == ord("q"):
+                    break
+            else:
+                break
+
+            i = i + 1
+        cam1_video.release()
+        cam2_video.release()
+        out.release()
+        cv2.destroyAllWindows()
+        print(f"Merged RGB video created at {trial_path}/{trial_name}_rgb.avi!")
+
+    def _splice_xma_to_dlc(
+        self, trial_path, outlier_mode=False, swap=False, cross=False
+    ):
+        """Takes csv of XMALab 2D XY coordinates from 2 cameras, outputs spliced hdf+csv data for DeepLabCut"""
+        substitute_data_relpath = "labeled-data/" + self._config["dataset_name"]
+        substitute_data_abspath = os.path.join(
+            os.path.sep.join(self._config["path_config_file"].split("\\")[:-1]),
+            substitute_data_relpath,
+        )
+        markers = self.get_bodyparts_from_xma(trial_path)
```

Comment:
Theoretically, this entire section can be solved with a creative call to get_bodyparts_from_xma and some dataFrame manipulation to be significantly shorter (and potentially faster?). Not something for this PR, but definitely something to note for later.